### PR TITLE
Feature/events

### PR DESCRIPTION
**Pull request by newnottakenname**

### CODE REVIEW

File: core/src/main/nl/tudelft/ti2806/riverrush/CoreModule.java [hunk 1]
```diff
@@ -36,6 +37,15 @@ protected Protocol configureRendererProtocol() {
...
 
+        protocol.registerNetworkAction(RenderJoinEvent.class, RenderJoinEvent::new);
+        protocol.registerNetworkAction(GameWaitingEvent.class, GameWaitingEvent::new);
+        protocol.registerNetworkAction(GameAboutToStartEvent.class, GameAboutToStartEvent::new);
+        protocol.registerNetworkAction(GameStartedEvent.class, GameStartedEvent::new);
+        protocol.registerNetworkAction(GameFinishedEvent.class, GameFinishedEvent::new);
+        protocol.registerNetworkAction(GameStoppedEvent.class, GameStoppedEvent::new);
+        protocol.registerNetworkAction(PlayerAddedEvent.class, PlayerAddedEvent::new);
+        protocol.registerNetworkAction(PlayerJumpedEvent.class, PlayerJumpedEvent::new);
+
```
Comment (txsmith) [inline core/src/main/nl/tudelft/ti2806/riverrush/CoreModule.java]: We should really make a method  to register multiple events with one handler

File: core/src/main/nl/tudelft/ti2806/riverrush/domain/event/PlayerJumpedEvent.java [hunk 2]
```diff
@@ -20,7 +16,7 @@ public String serialize(final Protocol protocol) {
 
     @Override
     public Event deserialize(final Map<String, String> keyValuePairs) {
-        this.player = new Player(Long.parseLong(keyValuePairs.get("player")));
+        //this.player = new Player(Long.parseLong(keyValuePairs.get("player")));
```
Comment (txsmith) [inline core/src/main/nl/tudelft/ti2806/riverrush/domain/event/PlayerJumpedEvent.java]: Deserializing the player is now broken, we should fix this ASAP

File: core/src/main/nl/tudelft/ti2806/riverrush/network/event/RenderJoinEvent.java [hunk 3]
```diff
@@ -10,12 +10,12 @@
 
     @Override
     public String serialize(Protocol protocol) {
-        return null;
+        return "key=value";
```
Comment (txsmith) [inline core/src/main/nl/tudelft/ti2806/riverrush/network/event/RenderJoinEvent.java]: This value is absolute BS, should fix this tomorrow

**Comment by mitchellolsthoorn** [comment]
Reviewed all code, looks good.

**Pull request merged**